assets/src/components/dev-tools-toggle/index.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress dependencies
+ */
+import { useContext } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { AMPSettingToggle } from '../amp-setting-toggle';
+import { User } from '../../components/user-context-provider';
+import { Loading } from '../loading';
+
+export function DevToolsToggle() {
+	const {
+		developerToolsOption,
+		fetchingUser,
+		setDeveloperToolsOption,
+	} = useContext( User );
+
+	if ( fetchingUser ) {
+		return <Loading />;
+	}
+
+	return (
+		<AMPSettingToggle
+			checked={ true === developerToolsOption }
+			title={ __( 'Enable Dev Tools', 'amp' ) }
```

Comment:
I don't think there's a need to use the shortened "Dev" form of "Developer", it may also cause confusion for some.

```suggestion
			title={ __( 'Enable Developer Tools', 'amp' ) }
```